### PR TITLE
strting format and negative indices

### PR DESCRIPTION
**Pull request by anharrington**
bigger individual changes than before.  See new subsection for string format method.

Also, a 10 second time limit for interactive loop programs is not enough; changed to 60 sec.

### CODE REVIEW

File: _sources/Strings/Exercises.rst [hunk 1]
```diff
@@ -875,3 +875,30 @@ Exercises
             .. disqus::
                 :shortname: interactivepython
                 :identifier: disqus_49e1151bb7864a3287a6b6ae1c84db16
+
+
+#. Modify this code so it prints each subtotal, the total cost, and average price
```
Comment (bnmnetp) [inline _sources/Strings/Exercises.rst]: This one will conflict with the change I pushed a little bit ago.

File: _sources/Functions/thinkcspyExercises.rst [hunk 2]
```diff
@@ -551,3 +551,13 @@ Exercises
             .. disqus::
                 :shortname: interactivepython
                 :identifier: db5d8808bf5749579718bdd2088b539f
+
```
Comment (anharrington) [inline _sources/Functions/thinkcspyExercises.rst]: I missed that!  I think I fixed a couple.  It looks like the lists Exercises still have a number of problems with that issue that I did not cause, I think.

File: _sources/Strings/StringMethods.rst [hunk 3]
```diff
@@ -145,10 +146,151 @@ change the original.  You can also consult the `Python documentation for strings
...
+
+A technical point: Since braces have special meaning in a format
+string, there must be a special rule if you want braces to actually
+be included in the final *formatted* string. The rule is to double
+the braces: ``{{`` and ``}}``. For example mathematical set
+notation uses braces. The initial and final doubled
+braces in the format string below generate literal braces in the
+formatted string:
+
+.. activecode:: ch08_methods7
```
Comment (bnmnetp) [inline _sources/Strings/StringMethods.rst]: I love the addition of the format section.  In running through the book with this example, I did notice there was a bug in skulpt that may be worth mentioning.  The example you use prints out `{{5, 9}}` in Skulpt.  Even more confusingly removing the first { and the last } give the result you want in Skulpt, but cause an error in CPython.

Comment (anharrington) [inline _sources/Strings/StringMethods.rst]: just pushed a version with a comment.  I put the example text in a code block, not active code block.  Can my pushes now get merged?

**Pull request merged**